backend/app/services/crawler.py

```
@@ -24,25 +51,124 @@ def _load_crawler_module():
+        if finished_at is not None:
+            _state["finished_at"] = finished_at
+
+
+def _run_pipeline_script(
+    script_name: str,
+    step: SyncStep,
+    message: str,
+    *args: str,
+) -> None:
+    _set_state(step=step, message=message)
+    subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)